src/IdObfuscator/IdObfuscator.php

```diff
@@ -0,0 +1,69 @@
+<?php
+namespace Craftwork\IdObfuscator;
+
+final class IdObfuscator
+{
+    const CHARSET_DEFAULT = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890-_';
+
+    /**
+     * @var string
+     */
+    private $characterSet;
+
+    /**
+     * @var int
+     */
+    private $characterSetLength;
+
+    /**
+     * @param string $salt Salt that should be unique and kept secret for your application.
+     * @param string $characterSet An optional character set.
+     */
+    public function __construct(string $salt, string $characterSet = self::CHARSET_DEFAULT)
+    {
+        $this->characterSet = $this->shuffleCharacterSetWithSalt($characterSet, $salt);
+        $this->characterSetLength = strlen($this->characterSet);
+    }
+
+    /**
+     * @param int $number The number to encode.
+     * @return string
+     */
+    public function encode(int $number) : string
+    {
+        $hash = '';
+
+        do {
+            $hash = $this->characterSet[$number % $this->characterSetLength] . $hash;
+            $number = (int) ($number / $this->characterSetLength);
+        } while ($number > 0);
+
+        return $hash;
+    }
+
+    public function decode(string $string) : int
```

Comment:
I've stopped using them for PHP7 unless I want to provide any more information than what's already in the type signature. Maybe I could detail what the string is although I think it's obvious.